adagrams/game.py

```diff
@@ -1,11 +1,142 @@
+import random
+import copy
+
+ # Two Dimensional Array
+
+LETTER_POOL = [
+	['A', 'A', 'A', 'A', 'A', 'A', 'A', 'A', 'A'],
+	['B', 'B'], 
+	['C', 'C'], 
+	['D','D','D','D'],
+	['E','E', 'E', 'E', 'E', 'E', 'E', 'E', 'E', 'E', 'E', 'E'], 
+	['F', 'F'], 
+	['G', 'G', 'G'], 
+	['H', 'H'], 
+	['I', 'I', 'I', 'I', 'I', 'I', 'I', 'I', 'I'], 
+	['J'], 
+	['K'], 
+	['L', 'L', 'L', 'L'], 
+	['M', 'M'],
+	['N', 'N', 'N', 'N', 'N', 'N'],
+	['O', 'O', 'O', 'O', 'O', 'O', 'O', 'O'], 
+	['P', 'P'], 
+	['Q'], 
+	['R', 'R', 'R', 'R', 'R', 'R'], 
+	['S', 'S', 'S', 'S'], 
+	['T', 'T', 'T', 'T', 'T', 'T'], 
+	['U', 'U', 'U', 'U'], 
+	['V', 'V'], 
+	['W', 'W'], 
+	['X'], 
+	['Y', 'Y'],
+	['Z']
+]
+
+# helper function to create one big list of strings
+
+def get_one_list_of_letters():
+
+    total_letters_list = []
+    for elem in LETTER_POOL:
+        for char in elem:
+            total_letters_list.append(char)
+    return total_letters_list
+
 def draw_letters():
-    pass
+
+    total_letters_list = get_one_list_of_letters()
+
+    # start with an empty hand and initialize counter at 0
+    hand = []
+    i = 0
+    while i < 10:
+        # randomize letter choice
+        random_letter = total_letters_list[random.randint(0, 97)]
+        # ensure that random letter choice in hand is never greater than the max amount in the letter pool
+        # if less than the max amount then append to empty list which is our hand of letters
+        if hand.count(random_letter) < total_letters_list.count(random_letter):
+            hand.append(random_letter)
+        else:
+            # use continue to end this iteration and immediately start next iteration
+            continue
+        i += 1
+
+    return hand    
 
 def uses_available_letters(word, letter_bank):
-    pass
+
+    # made copy of list as to not modify the original copy
+    letter_bank_copy = copy.deepcopy(letter_bank)
+
+    # iterate through each letter in the word and make letters all uppercase
+    for letter in word.upper():
+
+        if letter in letter_bank_copy:
+            # remove letter from letter bank so it can't be reused
+            letter_bank_copy.remove(letter)
+        else:
+           return  False
+
+    return True 
```

Comment:
Nice use of the `deepcopy` function here, it lets you get a copy of the `letter_bank` without changing the object itself. This shows me that you have a good understanding of variables and memory. An alternative approach that doesn't mutate the original `letter_bank` is this here:
```python
def uses_available_letters(word, letter_bank):
    cap_word = word.upper()
    for letter in cap_word:
       if letter_bank.count(letter) == 0: 
          return False

       if cap_word.count(letter) > letter_bank.count(letter):
          return False
    
    return True
```
It is basically the same thing as what you had in wave 1, using the `count` function to count the instances of a letter in both the `word` and `letter_bank`.